### PR TITLE
fix: Delete bundle region provider.

### PR DESCRIPTION
**Pull request by sichanyoo**
## Issue \#

https://github.com/awslabs/aws-sdk-swift/issues/1102

## Description of changes

Deletes unnecessary region provider that fetches region from Bundle for iOS / tvOS / watchOS applications.

## New/existing dependencies impact assessment, if applicable

N/A

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dayaffe** [comment]
Happy to approve but can you provide more details on why its unnecessary? Was it unused?

**Comment by jbelkins** [comment]
@dayaffe We asked both Amplify and our tester community at-large, no one seems to be using this

File: Sources/Core/AWSClientRuntime/Regions/DefaultRegionResolver.swift [hunk 1]
```diff
@@ -13,7 +13,6 @@ public struct DefaultRegionResolver: RegionResolver {
 
     public init(fileBasedConfigurationProvider: @escaping FileBasedConfigurationProviding) throws {
         self.providers = [
-            BundleRegionProvider(),
             EnvironmentRegionProvider(),
             ProfileRegionProvider(fileBasedConfigurationProvider: fileBasedConfigurationProvider),
             try IMDSRegionProvider()
```
Comment (jbelkins) [inline Sources/Core/AWSClientRuntime/Regions/DefaultRegionResolver.swift:18]: After removing `BundleRegionProvider`, are the rest of these region providers complete & in the right order?

Comment (sichanyoo) [reply Sources/Core/AWSClientRuntime/Regions/DefaultRegionResolver.swift:18]: Yes. 
`nY63AaBMJcoU/Cross-SDK-Feature-Matrix-Items-Deprecated-or-Without-SEPs#temp:C:UcSa0a343b6b2034712914a2965f`

**Comment by sichanyoo** [comment]
> Happy to approve but can you provide more details on why its unnecessary -- was it unused?

@dayaffe It's not part of the standard chain of places to resolve region from (environment, profile, then IMDS), and it's not clear whether it's even used. Checked with Amplify and they also don't specify region from application Bundle. Checked with internal slack channel and no reply from anyone. Josh also said that based on his experience, it would be unlikely that you'd want to put region info into the application Bundle.

**Review by dayaffe — APPROVED** [review]

**Pull request merged**